### PR TITLE
Rewrite if-else-if-else to switch statements

### PR DESCRIPTION
**Pull request by sosiska**
This commit rewrites some if-else-if-else chains as a switch.
Based on Go style guide:
https://golang.org/doc/effective_go.html#switch

Found by https://go-critic.github.io/overview#ifElseChain-ref

### CODE REVIEW

**Comment by sosiska** [comment]
Ping

File: fontmaker/core/fontmaker.go [hunk 1]
```diff
@@ -230,12 +230,12 @@ func (f *FontMaker) MakeFontDescriptor(info TtfInfo) (string, error) {
 	fd += fmt.Sprintf("\tme.desc[1] =  gopdf.FontDescItem{ Key: \"Descent\", Val : \"%d\" }\n", descender)
 
 	// CapHeight
-	capHeight, err := info.GetInt64("CapHeight")
-	if err == nil {
+	switch capHeight, err := info.GetInt64("CapHeight"); err {
```
[the remark this answers is not in the document]
Comment (sosiska) [reply fontmaker/core/fontmaker.go]: OK, rejected in c21dc562f9931980a528dba74d8cbb3b93850d30.

**Comment by sosiska** [comment]
Ping

File: image_obj_parse.go [hunk 2]
```diff
@@ -280,9 +282,9 @@ func parsePng(f *bytes.Reader, info *imgInfo, imgConfig image.Config) error {
 			if err != nil {
 				return err
 			}
-		} else if string(typ) == "IEND" {
+		case "IEND":
 			break
```
Comment (oneplus1000) [inline image_obj_parse.go]: in old code this "break" stop for-loop (at line 226), but in the new code this "break" stop switch-case and not stop for-loop anymore.

Comment (sosiska) [reply image_obj_parse.go]: But i can see outer break in golang source:

Comment (oneplus1000) [reply image_obj_parse.go]: Thank you. :blush: for the pull request.
"outer-break" in my opinion it makes code harder to read ( not much but still harder ) so I think don't use "outer-break" should be better.

**Pull request merged**